src/analysis/normed_space/normed_group_quotient.lean

```diff
@@ -0,0 +1,489 @@
+    use [y - m, h],
+    simp },
+  { rintros ⟨y, h, rfl⟩,
+    use m + y,
+    simpa using h },
+end
+
+/-- The quotient norm is nonnegative. -/
+lemma quotient_norm_nonneg (S : add_subgroup M) : ∀ x : quotient S, 0 ≤ ∥x∥ :=
+begin
+  rintros ⟨m⟩,
+  change 0 ≤ ∥mk' S m∥,
+  apply real.lb_le_Inf _ (image_norm_nonempty _),
+  rintros _ ⟨n, h, rfl⟩,
+  apply norm_nonneg
+end
+
+/-- The quotient norm is nonnegative. -/
+lemma norm_mk_nonneg (S : add_subgroup M) (m : M) : 0 ≤ ∥mk' S m∥ :=
+quotient_norm_nonneg S _
+
+/-- The norm of the image of `m : M` in the quotient by `S` is zero if and only if `m` belongs
+to the closure of `S`. -/
+lemma quotient_norm_eq_zero_iff (S : add_subgroup M) (m : M) :
+  ∥mk' S m∥ = 0 ↔ m ∈ closure (S : set M) :=
+begin
+  have : 0 ≤ ∥mk' S m∥ := norm_mk_nonneg S m,
+  rw [← this.le_iff_eq, quotient_norm_mk_eq, real.Inf_le_iff],
+  simp_rw [zero_add],
+  { calc (∀ ε > (0 : ℝ), ∃ r ∈ (λ x, ∥m + x∥) '' (S : set M), r < ε) ↔
+        (∀ ε > 0, (∃ x ∈ S, ∥m + x∥ < ε)) : by simp [set.bex_image_iff]
+     ... ↔ ∀ ε > 0, (∃ x ∈ S, ∥m + -x∥ < ε) : _
+     ... ↔ ∀ ε > 0, (∃ x ∈ S, x ∈ metric.ball m ε) : by simp [dist_eq_norm, ← sub_eq_add_neg,
+                                                              norm_sub_rev]
+     ... ↔ m ∈ closure ↑S : by simp [metric.mem_closure_iff, dist_comm],
+    apply forall_congr, intro ε, apply forall_congr, intro  ε_pos,
+    rw [← S.exists_neg_mem_iff_exists_mem],
+    simp },
+  { use 0,
+    rintro _ ⟨x, x_in, rfl⟩,
+    apply norm_nonneg },
+  rw set.nonempty_image_iff,
+  use [0, S.zero_mem]
+end
+
+lemma norm_mk_lt {S : add_subgroup M} (x : (quotient S)) {ε : ℝ} (hε : 0 < ε) :
+  ∃ (m : M), quotient_add_group.mk' S m = x ∧ ∥m∥ < ∥x∥ + ε :=
+begin
+  obtain ⟨_, ⟨m : M, H : mk' S m = x, rfl⟩, hnorm : ∥m∥ < ∥x∥ + ε⟩ :=
+    real.lt_Inf_add_pos (bdd_below_image_norm _) (image_norm_nonempty x) hε,
+  subst H,
+  exact ⟨m, rfl, hnorm⟩,
+end
+
+lemma norm_mk_lt' (S : add_subgroup M) (m : M) {ε : ℝ} (hε : 0 < ε) :
+  ∃ s ∈ S, ∥m + s∥ < ∥mk' S m∥ + ε :=
+begin
+  obtain ⟨n : M, hn : mk' S n = mk' S m, hn' : ∥n∥ < ∥mk' S m∥ + ε⟩ :=
+    norm_mk_lt (quotient_add_group.mk' S m) hε,
+  erw [eq_comm, quotient_add_group.eq] at hn,
+  use [- m + n, hn],
+  rwa [add_neg_cancel_left]
+end
+
+/-- The quotient norm satisfies the triangular inequality. -/
+lemma quotient_norm_add_le (S : add_subgroup M) (x y : quotient S) : ∥x + y∥ ≤ ∥x∥ + ∥y∥ :=
+begin
+  refine le_of_forall_pos_le_add (λ ε hε, _),
+  replace hε := half_pos hε,
+  obtain ⟨m, rfl, hm : ∥m∥ < ∥mk' S m∥ + ε / 2⟩ := norm_mk_lt x hε,
+  obtain ⟨n, rfl, hn : ∥n∥ < ∥mk' S n∥ + ε / 2⟩ := norm_mk_lt y hε,
+  calc ∥mk' S m + mk' S n∥ = ∥mk' S (m +  n)∥ : by rw (mk' S).map_add
+  ... ≤ ∥m + n∥ : quotient_norm_mk_le S (m + n)
+  ... ≤ ∥m∥ + ∥n∥ : norm_add_le _ _
+  ... ≤ ∥mk' S m∥ + ∥mk' S n∥ + ε : by linarith
+end
+
+/-- The quotient norm of `0` is `0`. -/
+lemma norm_mk_zero (S : add_subgroup M) : ∥(0 : (quotient S))∥ = 0 :=
```

Comment:
```suggestion
lemma norm_mk_zero (S : add_subgroup M) : ∥(0 : quotient S)∥ = 0 :=
```
There are other places where the parenthesis can be removed (I am not completely sure it is a good idea, but the code looks nicer to me).